libs/model/src/policies/CommunityGoals.policy.ts

```diff
@@ -1,56 +1,78 @@
 import { Policy, command } from '@hicommonwealth/core';
-import { events } from '@hicommonwealth/schemas';
+import { CommunityGoalReached, events } from '@hicommonwealth/schemas';
 import { Op } from 'sequelize';
+import { z } from 'zod';
 import { SetReachedGoal } from '../aggregates/community';
 import { models } from '../database';
 import { systemActor } from '../middleware';
 
 const inputs = {
   CommunityJoined: events.CommunityJoined,
+  ThreadCreated: events.ThreadCreated,
 };
 
+async function findOpenGoals(
+  community_id: string,
+  type: 'members' | 'threads',
+) {
+  const goals = await models.CommunityGoalReached.findAll({
+    where: { community_id, reached_at: { [Op.is]: null } },
+    include: [
+      {
+        model: models.CommunityGoalMeta,
+        as: 'meta',
+        required: true,
+        where: { type },
+      },
+    ],
+  });
+  return goals.map((g) => g.toJSON());
+}
+
+async function setReachedGoal(
+  goals: Array<z.infer<typeof CommunityGoalReached>>,
+  target: number,
+) {
+  await Promise.all(
```

Comment:
I presume the amount of community goals is expected to be small so this shouldn't be a performance issue?

Updating the goals is not in a transaction. Will this cause issues if not why is it ok in this case?

Reply:
yeah, and this is only checking communities that are configured in quests with open goals

each goal is a separate transaction, but I guess we can also make the entire operation atomic... will revisit in next pR